bundles/org.openhab.binding.noolite/src/main/java/org/openhab/binding/noolite/internal/NooliteMTRF64Adapter.java

```diff
@@ -0,0 +1,114 @@
+
+import gnu.io.CommPortIdentifier;
+import gnu.io.SerialPort;
+import gnu.io.SerialPortEvent;
+import gnu.io.SerialPortEventListener;
+
+/**
+ *
+ * @author Petr Shatsillo - Initial contribution
+ *
+ */
+public class NooliteMTRF64Adapter {
+
+    private static Logger logger = LoggerFactory.getLogger(NooliteMTRF64Adapter.class);
+    DataInputStream in = null;
+    DataOutputStream out = null;
+    Thread watcherThread = null;
+    private static OutputStream output;
+    SerialPort serial;
+
+    public void connect(NooliteBridgeConfiguration config) throws Exception {
+        logger.debug("Opening serial connection to port {} with baud rate 9600...", config.serial);
+
+        CommPortIdentifier portIdentifier;
+        try {
+            portIdentifier = CommPortIdentifier.getPortIdentifier(config.serial);
+            serial = portIdentifier.open("openhab", 3000);
+
+            serial.setSerialPortParams(9600, SerialPort.DATABITS_8, SerialPort.STOPBITS_1, SerialPort.PARITY_NONE);
+
+            in = new DataInputStream(serial.getInputStream());
+            output = serial.getOutputStream();
+            serial.addEventListener(new SerialPortEventListener() {
+                @Override
+                public void serialEvent(SerialPortEvent event) {
+                    try {
+                        byte[] data = new byte[17];
+                        if (in.read(data) > 0) {
+                            logger.debug("Received data: {}", DatatypeConverter.printHexBinary(data));
+                            short count = 0;
+                            byte sum = 0;
+                            for (int i = 0; i <= 14; i++) {
+                                count += (data[i] & 0xFF);
+                            }
+                            sum = (byte) (count & 0xFF);
+
+                            logger.debug("sum is {} CRC must be {} receive {}", count, sum, data[15]);
+
+                            if (((data[0] & 0xFF) == 0b10101101) && ((data[16] & 0xFF) == 0b10101110)) {
+                                logger.debug("sum is {} CRC must be {} receive {}", count, sum, data[15]);
+                                if (sum == data[15]) {
+                                    logger.debug("CRC is OK");
+
+                                    logger.debug("Updating values...");
+                                    NooliteMTRF64BridgeHandler.updateValues(data);
+                                } else {
+                                    logger.debug("CRC is WRONG");
+                                }
+                            } else {
+                                logger.debug("Start/stop bits is wrong");
+                            }
+                        }
+                    } catch (IOException ex) {
+                        logger.debug("Error reading from serial port!", ex);
+                    }
+                }
+            });
+            serial.notifyOnDataAvailable(true);
+        } catch (Exception e) {
```

Comment:
Can this be a specific exception? And always log exceptions. You may never know if someone gets this error and if nothing is logged. It will just silently fail.